WordPress/Classes/Services/MediaService.m

```diff
@@ -190,6 +192,48 @@ - (void)uploadMedia:(Media *)media
                 failure:failureBlock];
 }
 
+- (NSError *)processMediaUploadError:(NSError *)error {
+    NSError *newError = error;
+    if (error.domain == WordPressComApiErrorDomain) {
+        NSString *errorMessage = [error localizedDescription];
+        NSInteger errorCode = error.code;
+        NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithDictionary:error.userInfo];
+        switch (error.code) {
+            case WordPressComApiErrorUploadFailed:{
+                errorMessage = NSLocalizedString(@"The app couldn't upload this media.", @"Message to show to user when media upload failed by unknow server error");
+                errorCode = WordPressComApiErrorUploadFailed;
+            } break;
```

Comment:
We aren't explicit with styling of case blocks in our style guide but I want to say this isn't the norm across the app. You really don't need to local scope those two variable assignments.

```objc
case WordPressComApiErrorUploadFailed:
    errorMessage = NSLocalizedString(@"The app couldn't upload this media.", @"Message to show to user when media upload failed by unknow server error");
    errorCode = WordPressComApiErrorUploadFailed;
    break;
```